### PR TITLE
Use delta updates when updating mod list

### PR DESCRIPTION
**Pull request by jg18**
Addresses issue #154  and based on the solution proposed there. Still a bit of a WIP but tests are promising.

~~Known issue: the first time you switch from the Home tab to the Explore tab, there's that same 2-3 second delay that there currently is, but switching to Explore is fast after that. The reason is that I haven't (yet?) added support for warming the Explore cache at Knossos startup. I'll need to figure out a relatively clean way to warm the cache using the existing code paths.~~

EDIT: new commit initializes the explore mod list cache at startup. There is still a slight delay the first time you switch from the Home tab to the Explore tab but it's noticeably shorter than before and about as short as I can make it without per-field deltas which is not worth the extra work/difficulty IMO

### CODE REVIEW

**Comment by ngld** [comment]
In addition to the delta updates, we should remove a bunch of information which isn't necessary for the UI like the archives and file list. They make up a large chunk of data and never appear in the UI.
We'll need to be careful about the dev tab though since it needs some of the archive information (not the file list though).

I'll leave this open in case you want to try those changes, otherwise I'd gladly merge it.

**Comment by jg18** [comment]
Yeah I can work on implementing that.

File: html/templates/kn-page.vue [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default {
 
         search_text: '',
         status_message: '',
-        tab: 'explore',
+        tab: 'explore', // FIXME shouldn't this be initialized to 'home'? isn't that what's assumed elsewhere like in center?
```
Comment (jg18) [inline html/templates/kn-page.vue:24]: What should we do about this? I think that the "selected tab" highlighting appears on Explore at startup briefly before switching to Home when it should really be Home at the start. If you agree, I can either change it in this PR or you can change it in a separate commit.

**Comment by jg18** [comment]
Hmm, I can find the `filelist` field inside the `Package` class but can't find any fields named `archives` - what field(s) were you referring to?

File: html/templates/kn-page.vue (hunk 1 above)
Comment (ngld) [reply html/templates/kn-page.vue:24]: See https://github.com/ngld/knossos/pull/156#discussion_r251950324

**Comment by ngld** [comment]
Sorry, I meant `files` instead of `archives`. `files` was a bad name for that and I've already replaced it on the server with `archives` (which describes it much better) but forgot that this code still uses the old name.

**Pull request merged**